src/main/java/uk/gov/hmcts/reform/sendletter/services/StaleLetterService.java

```diff
@@ -0,0 +1,98 @@
+import uk.gov.hmcts.reform.sendletter.services.date.DateCalculator;
+
+import java.time.Clock;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.stream.Stream;
+
+import static java.time.temporal.ChronoUnit.DAYS;
+import static uk.gov.hmcts.reform.sendletter.util.TimeZones.EUROPE_LONDON;
+import static uk.gov.hmcts.reform.sendletter.util.TimeZones.UTC;
+
+@Service
+public class StaleLetterService {
+
+    private static final ZoneId SERVICE_TIME_ZONE_ID = ZoneId.of(EUROPE_LONDON);
+    private static final ZoneId DB_TIME_ZONE_ID = ZoneId.of(UTC);
+
+    private final DateCalculator dateCalculator;
+    private final LetterRepository letterRepository;
+    private final int minStaleLetterAgeInBusinessDays;
+    private final LocalTime ftpDowntimeStart;
+    private final Clock clock;
+
+    public StaleLetterService(
+        DateCalculator dateCalculator,
+        LetterRepository letterRepository,
+        @Value("${stale-letters.min-age-in-business-days}") int minStaleLetterAgeInBusinessDays,
+        @Value("${ftp.downtime.from}") String ftpDowntimeStart,
+        Clock clock
+    ) {
+        this.dateCalculator = dateCalculator;
+        this.letterRepository = letterRepository;
+        this.minStaleLetterAgeInBusinessDays = minStaleLetterAgeInBusinessDays;
+        this.ftpDowntimeStart = LocalTime.parse(ftpDowntimeStart);
+        this.clock = clock;
+    }
+
+    public Stream<StaleLetter> getStaleLetters() {
+        Stream<Letter> dbLetters = letterRepository.findStaleLetters(
+            calculateCutOffCreationDate()
+                .withZoneSameInstant(DB_TIME_ZONE_ID)
+                .toLocalDateTime()
+        );
+
+        return mapToStaleLetters(dbLetters);
+    }
+
+    /**
+     * Calculates the cut-off creation date for stale letters.
+     *
+     * @return Cut-off creation date for stale letters. Only letters older than that
+     *         can be considered stale.
+     */
+    private ZonedDateTime calculateCutOffCreationDate() {
+        ZonedDateTime now =
+            ZonedDateTime
+                .now(clock)
+                .withZoneSameInstant(SERVICE_TIME_ZONE_ID);
+
+        // If on a given day a letter was created within FTP downtime window or later,
+        // it should be treated like it was created the next day.
+        LocalTime ftpDowntimeAdjustedTime =
+            now.toLocalTime().isBefore(ftpDowntimeStart)
+                ? now.toLocalTime()
+                : ftpDowntimeStart;
+
+        ZonedDateTime todayWithFtpDowntimeAdjustedTime =
+            now
+                .truncatedTo(DAYS)
+                .plusNanos(ftpDowntimeAdjustedTime.toNanoOfDay());
+
+        return dateCalculator.subtractBusinessDays(
+            todayWithFtpDowntimeAdjustedTime,
+            minStaleLetterAgeInBusinessDays
+        );
+    }
+
+    private Stream<StaleLetter> mapToStaleLetters(Stream<Letter> dbLetters) {
```

Comment:
`map` function should operate with objects only instead of stream of objects which becomes a `flatmap`

Reply:
I'm not sure what you're referring to. Are you saying the use of `.map` is incorrect or the method name is wrong?

Reply:
technically a function which accepts `Stream` and returns `Stream` is a flatMap. But you are actually returning same `Stream but mapped over hence the function is named `mapSomething`. I'm suggesting to keep the map but function should be direct conversion from `Letter` to `StaleLetter` and amend calling place slightly to call `stream.map(this::mapToStaleLetters)` instead ;)

Reply:
`flatMap` is a combination of mapping and flattening - I don't see how this is happening here at all. Besides, we're not mapping a stream to stream, but letters to stale letters.

I'll do the direct mapping if that's your preference. It makes no difference from practical perspective.

Reply:
i can read the method yes it is not flatmapping anything :). just interpreting method signature here. and besides, this was not a pr blocker - stream closures are